.github/workflows/build_and_test.yml

```diff
@@ -41,8 +47,13 @@ jobs:
   build:
     name: "Build modules (${{ format('{0}, {1} job', needs.configure-jobs.outputs.branch, needs.configure-jobs.outputs.type) }}): ${{ matrix.modules }} ${{ matrix.comment }} (JDK ${{ matrix.java }}, ${{ matrix.hadoop }}, ${{ matrix.hive }})"
     needs: configure-jobs
-    # Do not run as scheduled jobs in forked repos
-    if: github.repository == 'apache/spark' || needs.configure-jobs.outputs.type == 'regular'
+    # Run scheduled jobs for Apache Spark only
+    # Run regular jobs for commit in both Apache Spark and fored repository
+    if: >-
+      (github.repository == 'apache/spark' &&
+        (needs.configure-jobs.outputs.type == 'scheduled' ||
+          needs.configure-jobs.outputs.type == 'regular'))
```

Comment:
`needs.configure-jobs.outputs.type == 'regular'` looks redundant? Below there is `|| needs.configure-jobs.outputs.type == 'regular'`.